### PR TITLE
Exclude unrequired mode for needs_top decision in get_intra_edges

### PR DESCRIPTION
**Pull request by takehirokj**
The output is same and encode time is almost same as [AWCY](https://beta.arewecompressedyet.com/?job=remove_unrequired_mode_in_selecting_part%402020-01-25T06%3A24%3A27.450Z&job=master-8ebe9233f6fbea4b3c83dd8f76129d196082e3d7).

### CODE REVIEW

File: src/partition.rs [hunk 1]
```diff
@@ -563,16 +563,9 @@ pub fn get_intra_edges<T: Pixel>(
...
-        || mode == PredictionMode::D135_PRED
-        || mode == PredictionMode::D113_PRED
-        || mode == PredictionMode::D157_PRED
-        || mode == PredictionMode::D203_PRED
-        || mode == PredictionMode::D67_PRED
-        || mode == PredictionMode::PAETH_PRED;
-      needs_top = !dc_or_cfl || y != 0;
+      needs_topleft =
+        mode.is_directional() || mode == PredictionMode::PAETH_PRED;
+      needs_top = (!dc_or_cfl || y != 0) && mode != PredictionMode::D203_PRED;
```
Comment (rzumer) [inline src/partition.rs]: > && mode != PredictionMode::D203_PRED
Why is this added?

Comment (takehirokj) [reply src/partition.rs]: This angle mode is not required for "needs_top" and this does not affect to the output.

Comment (rzumer) [reply src/partition.rs]: Can you specify where this is mentioned, e.g. spec section, page, or reference to libaom?

File: src/partition.rs [hunk 2]
```diff
@@ -563,16 +563,9 @@ pub fn get_intra_edges<T: Pixel>(
...
-        || mode == PredictionMode::H_PRED
-        || mode == PredictionMode::D45_PRED
-        || mode == PredictionMode::D135_PRED
-        || mode == PredictionMode::D113_PRED
-        || mode == PredictionMode::D157_PRED
-        || mode == PredictionMode::D203_PRED
-        || mode == PredictionMode::D67_PRED
-        || mode == PredictionMode::PAETH_PRED;
-      needs_top = !dc_or_cfl || y != 0;
+      needs_topleft =
```
Comment (rzumer) [inline src/partition.rs]: This is good to simplify now, but not all directional modes need the top-left pixel if IEF is disabled. Since I am planning to fix the top-left pixel pre-processing for IEF and put it in this function it will probably end up being reverted.

Comment (takehirokj) [reply src/partition.rs]: OK. I understand this simplification need to be changed after fixing #2107.
It is better to keep it as now.

File: src/partition.rs (hunk 1 above)
Comment (takehirokj) [reply src/partition.rs]: "AboveRow" is not required "if pAngle is greater than 180" as
in, 7.11.2.4.Directional intra prediction process (9.)
[AV1 spec](https://aomediacodec.github.io/av1-spec/av1-spec.pdf)

File: src/partition.rs (hunk 2 above)
Comment (takehirokj) [reply src/partition.rs]: This part is changed to original and change for "needs_top" is kept.
